helper/schema/field_reader_map.go

```diff
@@ -91,6 +92,57 @@ func (r *MapFieldReader) readPrimitive(
 	}, nil
 }
 
+func (r *MapFieldReader) readList(
+	address []string, schema *Schema) (FieldReadResult, error) {
+
+	addrPadded := make([]string, len(address)+1)
+	copy(addrPadded, address)
+
+	// Get the number of elements in the list
+	addrPadded[len(addrPadded)-1] = "#"
```

Comment:
Oh nm, it's a TypeList in the MapFieldReader. Saw "Map" and assumed TypeMap. All good!